server/shared/publishing/helpers.js

```diff
@@ -151,23 +151,39 @@ function publishAssessments(parent) {
 
 function fetchContainers(parent) {
   const typeConfigs = getSupportedContainers(parent.type);
-  const isCore = it => !containerRegistry.getStaticsResolver(it.type);
-  const coreTypes = typeConfigs.filter(isCore).map(it => it.type);
+  const { defaults, custom } = getGroupedContainerTypes(typeConfigs);
 
-  return Promise.all([
-    parent.getChildren({ where: { type: coreTypes } }).map(fetchDefaultContainer),
-    fetchCustomContainers(parent)
-  ])
-  .reduce((containers, groupedContainers) => {
+  const fetches = [];
+  if (defaults.length) fetches.push(fetchDefaultContainers(parent, defaults));
+  if (custom.length) fetches.push(fetchCustomContainers(parent, custom));
+  if (!fetches.length) return [];
+  return Promise.all(fetches).reduce((containers, groupedContainers) => {
     const mappedContainers = groupedContainers.map(it => {
       const config = find(typeConfigs, { type: it.type });
       const publishedAs = get(config, 'publishedAs', 'container');
-      return { ...it, publishedAs };
+      const { templateId } = config;
+      return { ...it, templateId, publishedAs };
     });
     return containers.concat(mappedContainers);
   }, []);
 }
 
+function getGroupedContainerTypes(typeConfigs) {
```

Comment:
Confusing name/doc might be helpful. typeConfigs 🤔 Feels like it might be nicer if we just pass activity and rename method to something like getActivityContainerTypes ?

Reply:
You are right. I'll make that change.